### PR TITLE
[serve] deflake test gcs failure

### PR DESCRIPTION
**Pull request by zcin**
## Why are these changes needed?

Make sure that proxy has been updated with the expected set of replicas before killing the GCS server in `test_new_router_on_gcs_failure` and `test_proxy_router_updated_replicas_then_gcs_failure`.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by zcin** [comment]
@edoakes @GeneDer ptal!

File: python/ray/serve/_private/proxy.py [hunk 1]
```diff
@@ -1293,6 +1294,10 @@ def _get_logging_config(self) -> Tuple:
                 log_file_path = handler.baseFilename
         return log_file_path
 
+    def _dump_ingress_replicas_for_testing(self, route) -> Set[ReplicaID]:
```
Comment (GeneDer) [inline python/ray/serve/_private/proxy.py:1297]: non-blocker: I kinda feel the way this is currently used, we don't really need to filter the replica by the route. We can probably just directly using `_dump_replica_states_for_testing()` from the controller? 

But also know that this is just refactored from the existing test, it's totally fine by me

Comment (GeneDer) [reply python/ray/serve/_private/proxy.py:1297]: nit: 
```suggestion
    def _dump_ingress_replicas_for_testing(self, route: str) -> Set[ReplicaID]:
```

**Review by GeneDer — APPROVED** [review]
LGTM!

File: python/ray/serve/_private/proxy.py (hunk 1 above)
Comment (zcin) [reply python/ray/serve/_private/proxy.py:1297]: `_dump_replica_states_for_testing` from the controller will give the list of replicas that the controller sees, but to deflake the test we need to make sure the replicas have been broadcasted to the proxy.

**Review by edoakes — APPROVED** [review]

**Pull request merged**